Fix a division-by-zero crash in the VPN code

diegoreymendez opened the pull request:
Task/Issue URL: https://app.asana.com/0/1203137811378537/1207299387361586/f

iOS PR: https://github.com/duckduckgo/iOS/pull/2862
macOS PR: https://github.com/duckduckgo/macos-browser/pull/2785
What kind of version bump will this require?: Patch

## Description

Fixes a crash caused by a division by zero.

## Testing

See platform-specific PRs for testing instructions.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

diegoreymendez commented on Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionBandwidthAnalyzer.swift:
```diff
@@ -111,8 +111,16 @@ final class NetworkProtectionConnectionBandwidthAnalyzer {
+        let tx: Double
+
+        if deltaSeconds > 0 {
+            rx = Double(newer.rxBytes - older.rxBytes) / deltaSeconds
+            tx = Double(newer.txBytes - older.txBytes) / deltaSeconds
```
Don't want to divide by zero.

mallexxx commented on Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionBandwidthAnalyzer.swift:
```diff
@@ -111,8 +111,16 @@ final class NetworkProtectionConnectionBandwidthAnalyzer {
 
     private func bytesPerSecond(newer: Snapshot, older: Snapshot) -> (rx: Double, tx: Double) {
         let deltaSeconds = newer.date.timeIntervalSince1970 - older.date.timeIntervalSince1970
```
could this be replaced with `newer.date.timeIntervalSince(older.date)` for better readability?

mallexxx commented on Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionBandwidthAnalyzer.swift:
```diff
@@ -111,8 +111,16 @@ final class NetworkProtectionConnectionBandwidthAnalyzer {
 
     private func bytesPerSecond(newer: Snapshot, older: Snapshot) -> (rx: Double, tx: Double) {
```
Since there was a crash related to this, I‘d suggest converting this function into `static` (as it‘s not using the state) and adding a test for it

mallexxx reviewed (approved):
Probably that should work, unless you do a horse kick test 😀

```
        let d = Date()
        let (rx, tx) = NetworkProtectionConnectionBandwidthAnalyzer.bytesPerSecond(newer: .init(rxBytes: 9_999_999_999_999, txBytes: 9_999_999_999_999, date: d.addingTimeInterval(0.0000001)),
                                                                                   older: .init(rxBytes: 0, txBytes: 0, date: d))

        let rxThreshold = 100 * 1024 // 100k
        let txThreshold = 100 * 1024 // 100k

        let uint_rx = UInt64(rx)
        let uint_tx = UInt64(tx)
```

But probably this condition is not something that has really happened, because lesser numbers work. Probably the crash was really about the division by zero leading to a NaN to UInt64 conversion.
Anyways, consider adding couple of small sanity unit tests

diegoreymendez commented:
I made the method static and added some tests to validate all is good and prevent regressions in the future.  I also changed the line that was suffering in readability, I agree.  Thanks for the suggestions.

The only suggestion I'm skipping is the larger inline code suggestion in your comment because IMO it favors a lower line count in detriment of readability and maintainability.

I want to make sure the reader can skim through the code and understand quickly, and I think this line isn't good to that end:

```swift
let (rx, tx) = NetworkProtectionConnectionBandwidthAnalyzer.bytesPerSecond(newer: .init(rxBytes: 9_999_999_999_999, txBytes: 9_999_999_999_999, date: d.addingTimeInterval(0.0000001)),
                                                                                   older: .init(rxBytes: 0, txBytes: 0, date: d))
```
